### PR TITLE
feat: add environment variables for host and port

### LINKED ISSUE #1326: [traces] make trace / app port configurable via env_var
Make the phoenix port configurable via env var `PHOENIX_PRORT` - the reason this is important is because now it's very easy to clash with another collector. This will allow two instances of phoenix to run at the same time making the DX much better. We can also force close old processes

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #1326

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -33,7 +33,6 @@ dependencies = [
   "pyarrow",
   "typing_extensions",
   "scipy",
-  "portpicker",
```
Comment (axiomofjoy) [inline pyproject.toml:36]: Do we not want to pick an open port by default if the user does not specify a port and the default port is empty?

**Review by mikeldking — APPROVED** [review]
Nice

**Review by axiomofjoy — APPROVED** [review]
LGTM. Can you make a ticket to update documentation around these changes?

File: pyproject.toml (hunk 1 above)
Comment (RogerHYang) [reply pyproject.toml:36]: yea, because app and exporter need to be in sync, i couldn't think of a good way to do that using random ports

Comment (axiomofjoy) [reply pyproject.toml:36]: Good point.

**Pull request merged**